internal/server/internal/ecdsa/ecdsa.go

```diff
@@ -0,0 +1,308 @@
+package ecdsa
+
+import (
+	"crypto"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"encoding/asn1"
+	"errors"
+	"io"
+	"math/big"
+)
+
+type rand struct {
+	kInv, r *big.Int
+}
+
+// NOTE: This buffer is designed to ensure that, regardless of how code outside
+// of this package uses these types and methods, no random value will ever be
+// used more than once in an ECDSA signature. When modifying this code, be VERY
+// CAREFUL that this behavior is maintained.
+
+// A RandProvider is capable of producing random values for use in ECDSA
+// signatures. It cannot be implemented by types outisde of this package.
+type RandProvider interface {
```

Comment:
We want this to be exported - it needs to be used outside the `ecdsa` package. But the `ecdsa` package is inside `internal`, so it's only accessible by other packages in `github.com/cloudflare/gokeyless`.